server/src/main/java/org/elasticsearch/action/search/SearchResponseMerger.java

```diff
@@ -93,6 +101,7 @@ final class SearchResponseMerger {
      */
     void add(SearchResponse searchResponse) {
         assert searchResponse.getScrollId() == null : "merging scroll results is not supported";
+        searchResponse.mustIncRef();
```

Comment:
`getMergedResponse` creates a new SearchResponse object when called. Does it also need to be incRef and decRef'd?  (If no, can you explain why - is it because it was not created in the Transport layer?)

Reply:
Yes the result needs to be properly ref counted as well. We want to reuse the same buffers that go into the merging for it. I may not have caught all the spots where the result has to be handled properly in this PR. I'd fix the remaining spots in a follow-up. This really is about fixing the merger itself (I only fixed a couple of obvious spots around those merge results here that naturally came out of my bigger PR that has this all working already :)).

Reply:
OK, I have a PR that is blocked on getting the SearchResponseMerger and TransportSearchAction ref-counting finished. In the new PR we are going to be calling getMergeResponse() more often - once for every SearchResponse from a cluster (for CCS minimize_roundtrips=true), so I'm going to close my current PR on that and start over once these changes are done.